pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -702,6 +702,11 @@ protected static PartitionedTopicMetadata fetchPartitionedTopicMetadata(PulsarSe
             return pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).get();
         } catch (Exception e) {
             if (e.getCause() instanceof RestException) {
+                // Since CompletableFuture wrappers exception, so we will lost the Status
+                // rebuild the Status if exception message contains "Topic not exist."
+                if (e.getCause().getMessage().contains("Topic not exist.")) {
```

Comment:
Please do not rely on strings, can we test the actual class?

Reply:
Yes, I'll change this PR and push again.